acceptance/tests/pxp-module-puppet/run_puppet_twice.rb

```diff
@@ -0,0 +1,95 @@
+require 'pxp-agent/test_helper.rb'
+require 'puppet/acceptance/environment_utils'
+
+test_name 'Run Puppet while a Puppet Agent run is in-progress, wait for completion' do
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  env_name = test_file_name = File.basename(__FILE__, '.*')
+  environment_name = mk_tmp_environment(env_name)
+
+  step 'On master, create a new environment that will result in a slow run' do
+    site_manifest = "#{environmentpath}/#{environment_name}/manifests/site.pp"
+    create_remote_file(master, site_manifest, <<-SITEPP)
+node default {
+  case $::osfamily {
+    'windows': { exec { 'sleep':
+                        command => 'true',
+                        unless  => 'sleep 10', #PUP-5806
+                        path    => 'C:\\cygwin64\\bin',} }
+    default:  { exec { '/bin/sleep 10': } }
+  }
+}
+SITEPP
+    on(master, "chmod 644 #{site_manifest}")
+  end
+
+  step 'Ensure each agent host has pxp-agent running and associated' do
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      create_remote_file(agent, pxp_agent_config_file(agent), pxp_config_json_using_puppet_certs(master, agent).to_s)
+      on agent, puppet('resource service pxp-agent ensure=running')
+      show_pcp_logs_on_failure do
+        assert(is_associated?(master, "pcp://#{agent}/agent"),
+               "Agent #{agent} with PCP identity pcp://#{agent}/agent should be associated with pcp-broker")
+      end
+    end
+  end
+
+  step 'Send two requests to all agents, the first with a long-running task' do
+    target_identities = []
+    agents.each do |agent|
+      target_identities << "pcp://#{agent}/agent"
+    end
+
+    provisional_responses = rpc_non_blocking_request(master, target_identities,
```

Comment:
This should maybe have a begin/rescue block so that any error here results in a clean beaker failure instead of an unhandled error.
Or, if a begin/rescue for every rpc call in every acceptance test seems excessive, consider removing the one at lines 51-58 - someone reading this code might wonder why the blocking request gets a rescue but the non-blocking doesn't